code/xcvm/lib/core/src/asset.rs

```diff
@@ -165,6 +165,10 @@ impl Amount {
 		self.slope.0 == 0
 	}
 
+	pub const fn is_both(&self) -> bool {
```

Comment:
This is a terrible name for the method.

Reply:
like is complex, is dual, is function, is linear, is absolute and ratio?

Reply:
Another option is to just get rid of it and explicitly test both. I think this type is convoluted enough that maybe it’s not worth making convenience methods whose meaning isn’t immediately obvious.